### DISCUSSION TITLE
REST API documentation?

### DISCUSSION THREAD

**Discussion by alaingaudreau** [minio/minio#16106] [Q&A]
Hello,

I thought I had seen API documentation for the admin rest api at some point, has this been removed?

I need to create users, groups, policies and buckets to interface with one of our client control panels and have only been able to find the madmin-go project which doesn't make for quick reading to pull out the various api calls it is making to /minio/admin/v3/ for those various methods.

I've seen broken links on other threads to the old madmin project which apparently contained documentation but it has since been removed.

Writing a wrapper for madmin-go for maybe a dozen api calls that we really need seems a bit tedious and error prone for nothing.

I understand, as it has been mentioned plenty of times before that you don't plan on building Admin SDK's for anything but Go which is fine but basic documentation on the Rest API that you are using behind that project would make integration into control panels so much simpler.

**Comment by harshavardhana** [comment]
https://pkg.go.dev/github.com/minio/madmin-go?utm_source=godoc

**Reply by alaingaudreau** [reply]
That's the madmin-go go api documentation. 

Not the minio v3 rest web api documentation for the endpoint(s) exposed at /minio/admin/v3/  the question is about to allow us to create users, groups and policies from a client control panel (cpanel / ispconfig / custom solutions, etc....) that is not built with Go and thus cannot use madmin-go.

**Reply by klauspost** [reply]
That is what is available. Also note there is no compatibility guarantees for the admin API.

While we don't have any non-open source code or documentation, we can offer direct to engineer support as part of our [subscription service and partner networks](https://min.io/pricing). Of course this also means that your system no longer is bound by AGPL requirements.
